.github/workflows/release.yml

```diff
@@ -28,55 +45,88 @@ jobs:
     steps:
       - name: Checkout the repo
         uses: actions/checkout@v3
+        with:
+          # `python-semantic-release` needs full history to properly determine the next release version
+          fetch-depth: 0
+
+      - name: Install poetry
+        run: pipx install poetry
+
+      - name: Configure poetry
+        run: |
+          poetry config virtualenvs.in-project true
+          poetry config repositories.testpypi https://test.pypi.org/legacy/
 
       - name: Set up Python
         uses: actions/setup-python@v3
         with:
           python-version: ${{ matrix.python-version }}
-
-      - name: Install poetry
-        run: pipx install poetry
+          cache: 'poetry'
 
       # TODO: Use a "CI" group to ensure only the dependencies needed for running in the CI environment are installed.
       #       This should be possible when Poetry 1.2 is available, using the groups feature:
       #       https://github.com/python-poetry/poetry/issues/1644
       #       https://github.com/phylum-dev/phylum-ci/issues/15
-      - name: Configure poetry
+      - name: Install the project with poetry
         run: |
-          poetry config virtualenvs.in-project true
           poetry env use python${{ matrix.python-version }}
           poetry install --verbose --no-root
 
-      # NOTE: This step will override, for this release only, the version set in the project. That is, the version will
-      #       be updated in the checked out source on the runner and used when building the package and release
-      #       artifacts but it will not be committed back to source control (e.g., in `main` branch). This means version
-      #       bumps in source control are not strictly enforced and syncronized with release versions.
-      # TODO: Consider a more fully automated continuous deployment pipeline. Some tools and workflows that can help are
-      #       using conventional commits, commitizen, commitlint, semantic-pull-requests, and python-semantic-release.
-      - name: Set the release version
-        run: poetry version ${GITHUB_REF#refs/tags/v}
-
-      # NOTE: Run the tests for the current active Python version, as a sanity check. It should uncover the situation
-      #       where the tag version doesn't adhere to the post-normalized PEP440 form: https://peps.python.org/pep-0440/
+      - name: Set phylum_original_version value
+        run: echo "phylum_original_version=$(poetry version --short)" >> $GITHUB_ENV
+
+      - name: Set to next version for build
+        run: |
+          if [ ${{ github.event.inputs.prerelease }} = "true" ]; then
+            poetry version $(poetry run semantic-release print-version --next --prerelease)
+          else
+            poetry version $(poetry run semantic-release print-version --next)
```

Comment:
What does `print-version` do if there haven't been any changes since the last release? Will this still bump the version number or will it be effectively a no-op.

Reply:
It prints `No release will be made` and works as a no-op in this context. The same question came up during the course of this work and here is the local testing that shows the behavior:

```sh
../phylum-ci  8  9 on  modern_release [!?⇡] is 📦 v0.1.1 via 🐍 v3.10.3
❯ poetry run semantic-release print-version --current
0.1.1%

../phylum-ci  8  9 on  modern_release [!?⇡] is 📦 v0.1.1 via 🐍 v3.10.3
❯ poetry run semantic-release print-version --next
0.2.0%

../phylum-ci  8  9 on  modern_release [!?⇡] is 📦 v0.1.1 via 🐍 v3.10.3
❯ poetry version $(poetry run semantic-release print-version --next)
Bumping version from 0.1.1 to 0.2.0

../phylum-ci  8  9 on  modern_release [!?⇡] is 📦 v0.2.0 via 🐍 v3.10.3
❯ poetry run semantic-release print-version --current
0.2.0%

../phylum-ci  8  9 on  modern_release [!?⇡] is 📦 v0.2.0 via 🐍 v3.10.3
❯ poetry run semantic-release print-version --next
No release will be made.

../phylum-ci  8  9 on  modern_release [!?⇡] is 📦 v0.2.0 via 🐍 v3.10.3
❯ poetry version $(poetry run semantic-release print-version --next)
No release will be made.
phylum 0.2.0

../phylum-ci  8  9 on  modern_release [!?⇡] is 📦 v0.2.0 via 🐍 v3.10.3
❯ poetry version -s
0.2.0

../phylum-ci  8  9 on  modern_release [!?⇡] is 📦 v0.2.0 via 🐍 v3.10.3
❯ poetry version $(poetry run semantic-release print-version --next).dev0
No release will be made.
```